near-sdk/src/promise.rs

```diff
@@ -253,6 +270,27 @@ impl Promise {
         self.add_action(PromiseAction::FunctionCall { function_name, arguments, amount, gas })
     }
 
+    /// A low-level interface for making a function call to the account that this promise acts on.
+    /// unlike [`Promise::function_call`], this function accepts a weight to use relative unused gas
+    /// on this function call at the end of the scheduling method execution.
+    pub fn function_call_weight(
+        self,
+        function_name: String,
+        arguments: Vec<u8>,
+        amount: Balance,
+        gas: Gas,
+        // TODO type
+        weight: u64,
```

Comment:
Yeah I think properly typing this would be a good idea

Reply:
Yeah, just was deciding between alias and newtype. Went with a newtype while keeping `with_unused_gas_weight` a u64 param for ergonomics